planner/core/rule_column_pruning.go

```diff
@@ -339,18 +339,12 @@ func (ds *DataSource) PruneColumns(parentUsedCols []*expression.Column, opt *log
 	if ds.schema.Len() == 0 {
 		var handleCol *expression.Column
 		var handleColInfo *model.ColumnInfo
-		if ds.table.Type().IsClusterTable() && len(originColumns) > 0 {
-			// use the first line.
-			handleCol = originSchemaColumns[0]
-			handleColInfo = originColumns[0]
+		// case 1: tiflash
+		if ds.tableInfo.TiFlashReplica != nil {
+			handleCol, handleColInfo = preferNotNullColumnFromTable(ds)
 		} else {
-			if ds.handleCols != nil {
-				handleCol = ds.handleCols.GetCol(0)
-				handleColInfo = handleCol.ToInfo()
-			} else {
-				handleCol = ds.newExtraHandleSchemaCol()
-				handleColInfo = model.NewExtraHandleColInfo()
-			}
+			// case 2: tikv
+			handleCol, handleColInfo = preferKeyColumnFromTable(ds, originSchemaColumns, originColumns)
```

Comment:
Add some comment like point to the `countStarRewriter` to illustrate the usage of the code block.